build_tools/python/e2e_test_artifacts/cmake_generator/iree_generator.py

```diff
@@ -0,0 +1,218 @@
+## Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Generates CMake rules to build IREE artifacts."""
+
+from dataclasses import dataclass
+from typing import Dict, List
+import pathlib
+
+from e2e_test_artifacts.cmake_generator import common_generators
+from e2e_test_artifacts import iree_artifacts
+from e2e_test_framework.definitions import common_definitions, iree_definitions
+import cmake_builder.rules
+
+
+@dataclass(frozen=True)
+class IreeModelImportRule(object):
+  target_name: str
+  output_file_path: str
+  cmake_rules: List[str]
```

Comment:
It looks like a rule that is more suitable for the `model_generator` module.

Reply:
This one is IREE specific rule, so I think it would be better to put it together with other IREE rules. `model_generator` will be used by other non-IREE benchmarks in the future, so we won't want them to import IREE rules.